### PR TITLE
CLDR-13803 st api: add a search API

### PR DESCRIPTION
**Pull request by srl295**
- plus tests
- search is asynchronous, and based on the CLDRFile API
- Exact matches only, for now

No UI… yet…

CLDR-13803

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/SearchAPI.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/SearchAPI.java:85]: what does "pre-validate response" mean?

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java:204]: should this be synchronized on searches or would that be superfluous?

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java:161]: java not formatted?

**Review by btangmu — COMMENTED** [review]
this looks great apart from formatting and a question or two

have we branched yet for v43? if not, should this PR be merged now, or wait?

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/SearchAPI.java
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/SearchAPI.java:85]: It means to check the response to see if it had a failure. No failure modes yet, so i'll remove the comment.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java:204]: superfluous.  the Cache is safe and only one thread has knowledge of that Search.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SearchManager.java:161]: I'll format it

**Comment by srl295** [comment]
> this looks great apart from formatting and a question or two
> 
> have we branched yet for v43? if not, should this PR be merged now, or wait?

have not branched . If i can get this ready it should go in.

**Comment by srl295** [comment]
@btangmu ready for review including UI

**Comment by srl295** [comment]
- need to be logged in and in a specific locale
- I shrunk the 'log out' button down to an icon
image
- search happens when you stop typing for half a second
- Search results show xpaths (paginated if there are lots of results)
image
- search window stays open until you close it, and you can pop it open again to jump to a different result.

**Review by btangmu — APPROVED** [review]

**Comment by srl295** [comment]
@btangmu want to wait until after you do the VXML extraction and then merge?

**Pull request merged**
